src/encoded/tests/test_create_mapping.py

```diff
@@ -55,87 +57,220 @@ def test_create_mapping_item_order(registry):
+    assert cfg['SKIP'] is False
+    assert cfg['WIPE_ES'] is True
+    assert cfg['STRICT'] is True
+
+
+@patch('dcicutils.deployment_utils.compute_ff_prd_env', MagicMock(return_value='fourfront-green'))
 @patch('encoded.commands.create_mapping_on_deploy.get_my_env', MagicMock(return_value='fourfront-green'))
-def test_get_deployment_config_prod_new():
+def test_get_deployment_config_prod():
     """ Tests get_deployment_config in the new production case """
-    cfg = get_deployment_config(None)
-    assert cfg['ENV_NAME'] == 'fourfront-green'  # sanity
-    assert cfg['WIPE_ES'] is False  # no wipe
+    my_env = create_mapping_on_deploy.get_my_env('ignored-for-mock')
+    assert my_env == 'fourfront-green'
+    with pytest.raises(RuntimeError):
+        CreateMappingOnDeployManager.get_deploy_config(env=my_env, args=MockedCommandArgs(), log=MockedLog())
 
 
-# It is essential to patch the place the function is referenced from, not the place it's defined from.
-# Sometimes both are needed, but not here. So we'll hold this other in reserve in case of a code change,
-# because these kinds of patches are sensitive to that. -kmp 28-Mar-2020
-#@patch('dcicutils.beanstalk_utils.whodaman', MagicMock(return_value='fourfront-webprod2'))
-@patch('encoded.commands.create_mapping_on_deploy.whodaman', MagicMock(return_value='fourfront-webprod2'))
-@patch('encoded.commands.create_mapping_on_deploy.get_my_env', MagicMock(return_value='fourfront-mastertest'))
-def test_get_deployment_config_mastertest_old():
-    """ Tests get_deployment_config in the mastertest case with an old-style ecosystem. """
-    cfg = get_deployment_config(None)
-    assert cfg['ENV_NAME'] == 'fourfront-mastertest'  # sanity
-    assert cfg['WIPE_ES'] is True  # wipe
-
-
-# It is essential to patch the place the function is referenced from, not the place it's defined from.
-# Sometimes both are needed, but not here. So we'll hold this other in reserve in case of a code change,
-# because these kinds of patches are sensitive to that. -kmp 28-Mar-2020
-#@patch('dcicutils.beanstalk_utils.whodaman', MagicMock(return_value='fourfront-green'))
-@patch('encoded.commands.create_mapping_on_deploy.whodaman', MagicMock(return_value='fourfront-green'))
+@patch('dcicutils.deployment_utils.compute_ff_prd_env', MagicMock(return_value='fourfront-green'))
+@patch('encoded.commands.create_mapping_on_deploy.get_my_env', MagicMock(return_value='fourfront-hotseat'))
+def test_get_deployment_config_hotseat():
+    """ Tests get_deployment_config in the hotseat case with a new-style ecosystem. """
+    my_env = create_mapping_on_deploy.get_my_env('ignored-for-mock')
+    assert my_env == 'fourfront-hotseat'
+    cfg = CreateMappingOnDeployManager.get_deploy_config(env=my_env, args=MockedCommandArgs(), log=MockedLog())
+    assert cfg['ENV_NAME'] == my_env  # sanity
+    assert cfg['SKIP'] is True  # The other values don't matter if this is set.
+    # assert cfg['WIPE_ES'] is ...
```

Comment:
No need to check these since they are not relevant.

Reply:
Comments trimmed.